crates/phactory/src/prpc_service.rs

```diff
@@ -401,46 +389,40 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> Phactory<Platform>
             cached_resp.encoded_runtime_info = runtime_info.encode();
         }
 
-        if !skip_ra {
-            if let Some(cached_attestation) = &cached_resp.attestation {
-                const MAX_ATTESTATION_AGE: u64 = 60 * 60;
-                if refresh_ra
-                    || reset_operator
-                    || now() > cached_attestation.timestamp + MAX_ATTESTATION_AGE
-                {
-                    cached_resp.attestation = None;
-                }
-            }
-            if cached_resp.attestation.is_none() {
-                // We hash the encoded bytes directly
-                let runtime_info_hash =
-                    sp_core::hashing::blake2_256(&cached_resp.encoded_runtime_info);
-                info!("Encoded runtime info");
-                info!("{:?}", hex::encode(&cached_resp.encoded_runtime_info));
-
-                let (attn_report, sig, cert) =
-                    match self.platform.create_attestation_report(&runtime_info_hash) {
-                        Ok(r) => r,
-                        Err(e) => {
-                            let message = format!("Failed to create attestation report: {:?}", e);
-                            error!("{}", message);
-                            return Err(from_display(message));
-                        }
-                    };
-
-                cached_resp.attestation = Some(pb::Attestation {
-                    version: 1,
-                    provider: "SGX".to_string(),
-                    payload: Some(pb::AttestationReport {
-                        report: attn_report,
-                        signature: base64::decode(sig).map_err(from_display)?,
-                        signing_cert: base64::decode_config(cert, base64::STANDARD)
-                            .map_err(from_display)?,
-                    }),
-                    timestamp: now(),
-                });
+        if let Some(cached_attestation) = &cached_resp.attestation {
+            const MAX_ATTESTATION_AGE: u64 = 60 * 60;
+            if refresh_ra
+                || reset_operator
+                || now() > cached_attestation.timestamp + MAX_ATTESTATION_AGE
```

Comment:
The expiration time might no longer be suitable to use the hard-coded one when we switch to DCAP.

Reply:
I haven't check DCAP's, but I think this should be generic for all providers